feature-libs/user/profile/occ/adapters/occ-user-profile.adapter.ts

```diff
@@ -159,4 +164,16 @@ export class OccUserProfileAdapter implements UserProfileAdapter {
       this.converter.pipeableMany(TITLE_NORMALIZER)
     );
   }
+
+  protected appendCaptchaToken(currentHeaders: HttpHeaders): HttpHeaders {
```

Comment:
Instead of doing this in each adapter where we want to use captcha, lets move this to a "captcha" interceptor instead to provide the public key in the request (see `global-message` interceptors for examples).

Is it possible to tell if an endpoint is going to be captcha protected pre-emptively before the request? If so, we would be able to tell when we want to use captcha in the header without configuring each endpoint on the frontend. But I'm not sure if we could get away with it...